src/components/Main/MainView/MessageInput/use/textFieldMirror.ts

```diff
@@ -0,0 +1,128 @@
+const properties = [
+  'position:absolute;',
+  'overflow:auto;',
+  'word-wrap:break-word;',
+  'top:0px;',
+  'left:-9999px;'
+]
+
+// カーソルの位置に影響するCSSプロパティ
+const propertyNamesToCopy = [
+  'box-sizing',
+  'font-family',
+  'font-size',
+  'font-style',
+  'font-variant',
+  'font-weight',
+  'height',
+  'letter-spacing',
+  'line-height',
+  'max-height',
+  'min-height',
+  'padding-bottom',
+  'padding-left',
+  'padding-right',
+  'padding-top',
+  'border-bottom',
+  'border-left',
+  'border-right',
+  'border-top',
+  'text-decoration',
+  'text-indent',
+  'text-transform',
+  'width',
+  'word-spacing'
+]
+
+const calcMirrorStyle = (textField: HTMLInputElement | HTMLTextAreaElement) => {
+  const style = window.getComputedStyle(textField)
+  const props = [...properties]
+  if (textField.nodeName.toLowerCase() === 'textarea') {
+    props.push('white-space:pre-wrap;')
+  } else {
+    props.push('white-space:nowrap;')
+  }
+  propertyNamesToCopy.forEach(name =>
+    props.push(`${name}:${style.getPropertyValue(name)};`)
+  )
+  return props.join(' ')
+}
+
+const mirrorMap = new WeakMap<HTMLInputElement | HTMLTextAreaElement, Mirror>()
+
+/**
+ * 元の要素と同じ親要素に紐付いたDiv.
+ * caret位置取得後にDestoroyするのは呼び出し元の責任
+ */
+type Mirror = HTMLDivElement
+
+/**
+ * カーソルと同じ位置に存在するspanエレメント
+ */
+type Marker = HTMLSpanElement
+
+/**
+ * テキストフィールドを模したdiv要素を画面外に準備する
+ *
+ * @param textField HTMLInputElement or HTMLTextAreaElement
+ * @param markerPosition 再現したいカーソルのインデックス(デフォルト値: テキスト終端)
+ */
+const textFieldMirror = (
+  textField: HTMLInputElement | HTMLTextAreaElement,
+  markerPosition: number | null
+): { mirror: Mirror; marker: Marker } => {
```

Comment:
ごめん、`mirror`は親要素が一致してるから`HTMLTextAreaElement`が破壊されれば破壊されるから`destroy`しなくても問題なかったからこうで :pray: (あと`type`で切り出すの意図は`type AA = { mirror: HTMLDivElement, marker: HTMLSpanElement }`ってことだった 🙇 )

```suggestion
const mirrorMap = new WeakMap<HTMLInputElement | HTMLTextAreaElement, HTMLDivElement>()

/**
 * テキストフィールドのキャレットの位置を再現した要素を取得する
 * @param markerPosition 再現したいカーソルのインデックス(デフォルト値: テキスト終端)
 */
const textFieldMirror = (
  textField: HTMLInputElement | HTMLTextAreaElement,
  markerPosition: number | null
): HTMLSpanElement => {
```

この関数普通にテキストフィールド内のキャレットの位置返したほうがよかったりしない？

あと、このファイル別にリアクティブなもの扱ってないから`@/lib`のがいいと思う